spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+          'listeningPort': 80,
+          'maxUtilization': 0.8,
+          'capacityScaler': 0.8
+        },
+        'availabilityZones': {
+          bindings['TEST_GCE_REGION']: [bindings['TEST_GCE_ZONE']]
+        },
+        'instanceMetadata': {
+          'startup-script': ('sudo apt-get update'
+                             ' && sudo apt-get install apache2 -y'),
+          'global-load-balancer-names': self.__lb_name,
+          'backend-service-names': 'backend-service',
+          'load-balancing-policy': json.dumps(policy)
+        },
+        'account': bindings['GCE_CREDENTIALS'],
+        'authScopes': ['compute'],
+        'user': '[anonymous]'
+      }],
+      description='Create Server Group in ' + group_name,
+      application=self.TEST_APP
+    )
+
+    builder = gcp.GcpContractBuilder(self.gcp_observer)
+    (builder.new_clause_builder('Managed Instance Group Added',
+                                retryable_for_secs=30)
+     .inspect_resource('instanceGroupManagers', group_name)
+     .contains_path_eq('targetSize', 1)
+    )
+
+    return st.OperationContract(
+      self.new_post_operation(title='create server group',
+                              data=payload, path='tasks'),
+      contract=builder.build()
+    )
+
+
+  def delete_server_group(self):
+    """Creates OperationContract for deleteServerGroup.
+
+    To verify the operation, we just check that the GCP managed instance group
+    is no longer visible on GCP (or is in the process of terminating).
+    """
+    bindings = self.bindings
+    group_name = '{app}-{stack}-v000'.format(
+        app=self.TEST_APP, stack=bindings['TEST_STACK'])
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[{
+        'cloudProvider': 'gce',
+        'serverGroupName': group_name,
+        'region': bindings['TEST_GCE_REGION'],
+        'zone': bindings['TEST_GCE_ZONE'],
+        'asgName': group_name,
+        'type': 'destroyServerGroup',
+        'regions': [bindings['TEST_GCE_REGION']],
+        'zones': [bindings['TEST_GCE_ZONE']],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'user': '[anonymous]'
+      }],
+      application=self.TEST_APP,
+      description='DestroyServerGroup: ' + group_name
+    )
+
+    builder = gcp.GcpContractBuilder(self.gcp_observer)
+    (builder.new_clause_builder('Managed Instance Group Removed')
+     .inspect_resource('instanceGroupManagers', group_name,
+                       no_resource_ok=True)
+     .contains_path_eq('targetSize', 0))
+
+    (builder.new_clause_builder('Instances Are Removed',
+                                retryable_for_secs=30)
+     .list_resource('instances')
+     .excludes_path_value('name', group_name))
+
+    return st.OperationContract(
+      self.new_post_operation(
+        title='delete server group', data=payload, path='tasks'),
+      contract=builder.build()
+    )
+
+
+class GoogleHttpLoadBalancerTest(st.AgentTestCase):
```

Comment:
I dont see a difference between this and the other GoogleHttpLoadBalancerTest defined above.
Could you have one class derive from the other (probably that from this since that uses the scenario defined here) and have the specialized class just clarify the difference. Hopefully it is very small.